### ISSUE TITLE
Update to app store "The App Name you entered is already being used"

### ISSUE THREAD

**Issue by florisdipt** [fastlane/fastlane#18242]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I get the output "The provided entity includes an attribute with a value that has already been used on a different account. - The App Name you entered is already being used." while the upload succeeded and is visible in app store connect. This does not happen for testflight.
I've confirmed this by version bumping and reproducing. 

##### Command executed

bundle exec fastlane ios build_ios && bundle install && bundle exec fastlane ios deploy_ios testflight:false

##### Complete output when running fastlane, including the stack trace and command used

Was to long for issue created a gist:
https://gist.githubusercontent.com/florisdipt/ddd5f61ece6e2ef35318d8e4622da968/raw/63fd02ac28752e2ef35bc09c6d8421a4cbd7abd0/gistfile1.txt

### Environment

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.15.7                                                        |
| Ruby                        | 2.6.3                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.24.3 (Apple Git-128)                             |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.15.7 (19H2)                                        |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 11.6                                                           |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
platform :ios do
  desc "Build & sign iOS app"
  lane :build_ios do |options|
    match(type: "appstore")
    sh "./flutter_build.sh --clean"
    build_ios_app(
      export_options: {
          provisioningProfiles: ENV['MATCH_PROVISIONING_PROFILE_MAPPING'],
          method: "app-store"
      }
    )
  end

  desc "Build & sign iOS app adhoc"
  lane :build_ios_adhoc do |options|
    match(type: "adhoc",
      app_identifier: "#REDACTED#")
    sh "./flutter_build.sh --clean"
    build_ios_app(
      export_options: {
          provisioningProfiles: ENV['MATCH_PROVISIONING_PROFILE_MAPPING'],
          method: "ad-hoc"
      }
    )
  end

  desc "Upload iOS app to app store"
  lane :deploy_ios do |options|
    # Upload to test flight or AppStore depending on caller parameters
    if options[:testflight]
      match(type: "appstore", app_identifier: "#REDACTED#")
      upload_to_testflight(
        skip_submission: true,
        # skip_metadata: false,
        # skip_screenshots: false,
        # force: true,
        ipa: "./Runner.ipa"
      )
    else
      match(type: "appstore",
      app_identifier: "#REDACTED#")
      upload_to_app_store(
        skip_metadata: false,
        skip_screenshots: false,
        submit_for_review: false,
        force: true,
        team_id: "#REDACTED#",
        ipa: "./Runner.ipa"
      )
    end
  end
end
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier("#REDACTED#") # The bundle identifier of your app
apple_id("#REDACTED#") # Your Apple email address

itc_team_id("#REDACTED#") # App Store Connect Team ID
team_id("#REDACTED#") # Developer Portal Team ID

# For more information about the Appfile, see:
#     https://docs.fastlane.tools/advanced/#appfile

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.175.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                           | Version      |
| ----------------------------- | ------------ |
| did_you_mean                  | 1.3.0        |
| slack-notifier                | 2.3.2        |
| rouge                         | 2.0.7        |
| xcpretty                      | 0.3.0        |
| terminal-notifier             | 2.0.0        |
| unicode-display_width         | 1.7.0        |
| terminal-table                | 1.8.0        |
| addressable                   | 2.7.0        |
| multipart-post                | 2.0.0        |
| word_wrap                     | 1.0.0        |
| tty-screen                    | 0.8.1        |
| tty-cursor                    | 0.7.1        |
| tty-spinner                   | 0.9.3        |
| artifactory                   | 3.0.15       |
| colored                       | 1.2          |
| highline                      | 1.7.10       |
| commander-fastlane            | 4.4.6        |
| faraday_middleware            | 1.0.0        |
| gh_inspector                  | 1.1.3        |
| rubyzip                       | 2.3.0        |
| security                      | 0.1.3        |
| dotenv                        | 2.7.6        |
| bundler                       | 2.1.4        |
| simctl                        | 1.6.8        |
| uber                          | 0.1.0        |
| declarative                   | 0.0.20       |
| declarative-option            | 0.1.0        |
| representable                 | 3.0.4        |
| retriable                     | 3.1.2        |
| mini_mime                     | 1.0.2        |
| httpclient                    | 2.8.3        |
| google-api-client             | 0.38.0       |
| nanaimo                       | 0.3.0        |
| colored2                      | 3.1.2        |
| claide                        | 1.0.3        |
| CFPropertyList                | 3.0.3        |
| atomos                        | 0.1.3        |
| xcodeproj                     | 1.19.0       |
| plist                         | 3.6.0        |
| public_suffix                 | 4.0.6        |
| babosa                        | 1.0.4        |
| excon                         | 0.79.0       |
| unf_ext                       | 0.0.7.7      |
| unf                           | 0.1.4        |
| domain_name                   | 0.5.20190701 |
| http-cookie                   | 1.0.3        |
| ruby2_keywords                | 0.0.4        |
| faraday-net_http              | 1.0.1        |
| faraday                       | 1.3.0        |
| faraday-cookie_jar            | 0.0.7        |
| fastimage                     | 2.2.2        |
| json                          | 2.5.1        |
| mini_magick                   | 4.11.0       |
| naturally                     | 2.2.1        |
| xcpretty-travis-formatter     | 1.0.1        |
| jwt                           | 2.2.2        |
| multi_json                    | 1.15.0       |
| signet                        | 0.14.1       |
| os                            | 1.1.1        |
| memoist                       | 0.16.2       |
| googleauth                    | 0.15.1       |
| rake                          | 13.0.3       |
| digest-crc                    | 0.6.3        |
| webrick                       | 1.7.0        |
| rexml                         | 3.2.4        |
| google-apis-core              | 0.2.1        |
| google-apis-storage_v1        | 0.2.0        |
| google-apis-iamcredentials_v1 | 0.1.0        |
| google-cloud-errors           | 1.0.1        |
| google-cloud-env              | 1.4.0        |
| google-cloud-core             | 1.5.0        |
| google-cloud-storage          | 1.30.0       |
| emoji_regex                   | 3.2.1        |
| aws-eventstream               | 1.1.0        |
| aws-sigv4                     | 1.2.2        |
| aws-partitions                | 1.427.0      |
| jmespath                      | 1.4.0        |
| aws-sdk-core                  | 3.112.0      |
| aws-sdk-kms                   | 1.42.0       |
| aws-sdk-s3                    | 1.88.1       |
| forwardable                   | 1.2.0        |
| logger                        | 1.3.0        |
| date                          | 2.0.0        |
| stringio                      | 0.0.2        |
| zlib                          | 1.0.0        |
| ipaddr                        | 1.2.2        |
| openssl                       | 2.1.2        |
| ostruct                       | 0.1.0        |
| strscan                       | 1.0.0        |
| fileutils                     | 1.1.0        |
| etc                           | 1.0.1        |
| io-console                    | 0.4.7        |
| libxml-ruby                   | 3.1.0        |
| psych                         | 3.1.0        |
| mutex_m                       | 0.1.0        |
</details>

*generated on:* **2021-02-19**
</details>

**Comment by max-ott** [comment]
App names are unique. Somebody else is already using your app name. Please try to change it.

**Comment by florisdipt** [comment]
> App names are unique. Somebody else is already using your app name. Please try to change it.
This is a update to a existing app. I've also verified no other apps are using the name.

**Comment by florisdipt** [comment]
Does this come directly from the Apple servers, or from fastlane? If this is a direct API response from Apple i'll try to contact their support team.

**Comment by max-ott** [comment]
It's directly coming from Apple. What happens if you try to change something on App Store Connect and then hit save?

**Comment by florisdipt** [comment]
> It's directly coming from Apple. What happens if you try to change something on App Store Connect and then hit save?

Okay thanks for the information.
While waiting for the initial response to this issue I ended up uploading another testflight version and promoting that one to a release version. 
It indeed said the same thing when trying to submit on App store connect. (App name in use)
I was able to submit after changing one of my languages its app name. 

Weird thing is upon deploying it bumped the app version and put all the information in but it did not upload the binary to the app store. Which i thought it did before because I saw the version being bumped.
Is this desired/documented behavior? 
Imo it should upload the binary either way especially if it faults the app metadata with like in my case, because it did set the metadata which was non valid but did not set the valid binary.

For easy reading i'll drop this below

`match(type: "appstore",
      app_identifier: "#REDACTED#")
      upload_to_app_store(
        skip_metadata: false,
        skip_screenshots: false,
        submit_for_review: false,
        force: true,
        team_id: "#REDACTED#",
        ipa: "./Runner.ipa"
      )`
